src/lib/rest/StringFilter.cpp

```diff
@@ -846,6 +892,7 @@ bool StringFilter::parse(const char* q, std::string* errorStringP)
     }
 
     str = NULL;  // So that strtok_r continues eating the initial string
+    item.patternValueToBeFreed = false;
```

Comment:
Auto-answer: the parse() processing make change the field between L882 and this point in code.